generator/src/main/java/com/scottlogic/deg/generator/validators/GenerationConfigValidator.java

```diff
@@ -8,80 +8,116 @@
-            } else if (!fileUtils.isDirectory((FileOutputTarget)outputTarget)) {
-                errorMessages
-                    .add("Invalid Output - not a directory, please enter a valid directory name");
-            } else if (!configSource.overwriteOutputFiles() && !fileUtils
-                .isDirectoryEmpty((FileOutputTarget)outputTarget, ruleCount)) {
-                errorMessages.add(
-                    "Invalid Output - directory not empty, please remove any 'manfiest.json' and '[0-9].csv' files or use the --overwrite option");
+    private void checkProfileInputFile(ArrayList<String> errorMessages, File profileFile) {
+        if (fileUtils.containsInvalidChars(profileFile)) {
+            errorMessages.add(String.format("Profile file path (%s) contains one or more invalid characters ? : %% \" | > < ", profileFile.toString()));
+        }
+        else if (!profileFile.exists()){
+            errorMessages.add("Invalid Input - Profile file does not exist");
+        }
+        else if (profileFile.isDirectory()) {
+            errorMessages.add("Invalid Input - Profile file path provided is to a directory");
+        }
+        else if (fileUtils.isFileEmpty(profileFile)) {
+            errorMessages.add("Invalid Input - Profile file has no content");
+        }
+        else {
+            try {
+                String fileType = fileUtils.probeContentType(profileFile.toPath());
+                if (!fileType.equals("application/json")) {
+                    errorMessages.add(String.format("Invalid Input - File is of type %s\nFile type application/json required", fileType));
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+                errorMessages.add("Invalid Input - Unable to determine content type of profile file");
             }
         }
     }
 
+    private void checkGenerateOutputTarget(ArrayList<String> errorMessages,
+                                           GenerationConfigSource configSource,
+                                           FileOutputTarget outputTarget) {
+        if (fileUtils.isDirectory(outputTarget)) {
+            errorMessages.add(
+                "Invalid Output - target is a directory, please use a different output filename");
+        } else if (!configSource.overwriteOutputFiles() && fileUtils.exists(outputTarget)) {
+            errorMessages.add(
+                "Invalid Output - file already exists, please use a different output filename or use the --overwrite option");
+        }
+    }
+
+    private void checkViolationGenerateOutputTarget(ArrayList<String> errorMessages,
+                                                    GenerationConfigSource configSource,
+                                                    FileOutputTarget outputTarget,
+                                                    int ruleCount) {
+        if (!fileUtils.exists(outputTarget)) {
+            errorMessages.add(
```

Comment:
This is checking that a directory is represented by a `FileOutputTarget`, is this right?

Reply:
This particular line (`  if (!fileUtils.exists(outputTarget)) {`) checks to see if the output target exists (which for violation, the directory must already exist)